### ISSUE TITLE
Blurry owncloud status icon

### ISSUE THREAD

**Issue by alefa** [numixproject/numix-icon-theme#944]
Using the Numix icon theme with Gnome shell, I have this small issue: the status icon of the ownCloud client in the top bar looks blurry and strangely squashed. Since Github's file upload didn't work for me, I uploaded my screenshots to imgur. This is how it looks with Numix (cloud-shaped icon on the left): [owncloud numix](https://imgur.com/bmq7cNU)
And this is the default status icon that comes with the application: [owncloud default](http://i.imgur.com/DcW3S74.png)
Looks much better in my opinion. Maybe the numix icon only looks blurry on a HiDPI screen. I can't test that because I don't have a different screen.
Would it be possible to remove the blurry icon and simply use the default, or replace it by a better looking one? I wanted to delete the blurry icon myself, but couldn't find it under Numix/. Could you tell me where I can find it?

**Comment by Foggalong** [comment]
That's really weird. [This is what it should look like, so that's definitely something we want to fix.](https://cloud.githubusercontent.com/assets/5920259/11764389/3cc55ae6-a125-11e5-9a87-0c9de3eb4726.png)

**Comment by alefa** [comment]
Good to hear that. Tell me if I can be of any help.

**Comment by palob** [comment]
Do you use [the TopIcons extension](https://extensions.gnome.org/extension/495/topicons/)?
If yes, does it look the same when it's turned off?

**Comment by alefa** [comment]
Yes, I do use the TopIcons extension. With the extension turned off, the owncloud status icon looks like this: [imgur](https://imgur.com/GQ17CdS)
I'm using Gnome 3.14 where tray icons are displayed in the message tray at the bottom.

**Comment by dirtydancing** [comment]
@alefa What is the name of the status icon in question? To me, this situation looks like the original icon is *-symbolic-named, whereas the blurry icon might not be *symbolic-named? This is because the blurry icon has a darker colour, which might be because colours of non-symbolic-named icons do not change with the respective GTK theme.

**Comment by alefa** [comment]
The only status icon I could find in /usr/share/icons/Numix is "owncloud-client.svg". This is what I get when I search for "owncloud" in this directory and its subdirectories:

    carbon:Numix $ find . -type f -iname "*owncloud*"
    ./32/apps/goa-account-owncloud.svg
    ./24/apps/goa-account-owncloud.svg
    ./24/status/owncloud-client.svg
    ./22/apps/goa-account-owncloud.svg
    ./22/status/owncloud-client.svg
    ./64/apps/goa-account-owncloud.svg
    ./48/apps/goa-account-owncloud.svg
    ./16/apps/goa-account-owncloud.svg
    ./16/status/owncloud-client.svg

The original icon seems to be "owncloud.png" in /usr/share/icons/hicolor. None of the owncloud status icons on my system has "symbolic" in its file name.

**Comment by dirtydancing** [comment]
Well, then I am not sure what even the icon name in question is. Where does the icon with the check mark come from? Is this some kind of composite with an emblem? Maybe s.th. specific to owncloud? I am afraid I do not know what is going on here.

**Comment by bil-elmoussaoui** [comment]
The icon names are 
- state-error 
- state-information
- state-sync
- state-ok
- state-offline
- state-pause

**Comment by wa4557** [comment]
I would also argue that the icon from your screenshot is not the "default" icon (I think it is not monochrome)? What's the icon theme you used in this screenshot?

**Comment by alefa** [comment]
@bil-elmoussaoui: Thanks, now I realize that I didn't find these icons because their file name doesn't contain "owncloud". Could the blurry look be due to the fact that these icons are only 24x24 pixels?

@wa4557: The icon from my screenshot is the default one when I select the option "Use monochrome icon" in the owncloud client's settings under "General". Without this option, the icon looks like this: [screenshot](https://imgur.com/2x9bXpl)
I took this screenshot with the default Gnome icon theme.

**Comment by wa4557** [comment]
@dirtydancing @bil-elmoussaoui To you have this issue as well? (I think both of you are using gnome-shell)

**Comment by dirtydancing** [comment]
Yeah, still happily using Arch/GNOME, but not using owncloud :)

**Comment by bil-elmoussaoui** [comment]
The icons do exists in other sizes too (16x16 and 22x22). Can you try an other GTK theme with Numix icons and see if you still have this issue? 
I do use Gnome-shell but i don't have this issue with other indicator icons. I will try to install OwnCloud and see if i can reproduce this. @alefa Which version of OwnCloud you're using ?

**Comment by dirtydancing** [comment]
But maybe I can check: do I install owncloud-client?

**Comment by dirtydancing** [comment]
OK, just checked with owncloud-client (not connecting apparently). In the "legacy" status icons row in the lower left corner, I can see the `state-offline` icon. This is GNOME Shell 3.18.3. Screenshot:

![owncloud](https://cloud.githubusercontent.com/assets/7164227/11839045/2c3b3c36-a3eb-11e5-9792-b26161d61713.png)

**Comment by bil-elmoussaoui** [comment]
@dirtydancing I have the same result.

**Comment by alefa** [comment]
My version of the owncloud client is 2.0.2. I'm using Debian Jessie with Gnome Shelll 3.14. Normally I use the Arc Gnome Shell and GTK theme, but I just checked that with Adwaita the status icon looks [identical](https://imgur.com/zgAqSji). As I mentioned before, I have a HiDPI screen. My laptop has 2560x1440 at 14'' and my external monitor 3840x2160 at 27''. I have set window scaling to 1 and font scaling to 1.6 in Gnome Tweak Tool.

**Comment by bil-elmoussaoui** [comment]
Disable using monochrome icons and try using Hardcode-Tray?

**Comment by dirtydancing** [comment]
owncloud client 2.1. here. No HiDPI. With TopIcons extension, I get the very same icon in the top bar as without TopIcons extension. And yes, depending on the GNOME Shell theme, all these "legacy" status icons might get somehwat blurry. It seems to be displaying fine, i.e. not blurry, at my end with default Adwaita GNOME Shell theme.

**Comment by alefa** [comment]
> Disable using monochrome icons and try using Hardcode-Tray?

@bil-elmoussaoui: Is this something you suggest me to try? When using the Numix icon theme, unchecking the "Use monochrome icon" option does not make a difference. The icon is always monochrome.
I don't know what Hardcode-Tray is and Google didn't tell me either.

**Comment by dirtydancing** [comment]
Just for completion: At my end, the original icon is displaying fine, just as the Numix one. Screenshot:

![owncloud_original](https://cloud.githubusercontent.com/assets/7164227/11841543/4e8022e4-a3fe-11e5-986b-2520d7ce050a.png)
 
Also fine in the GNOME Shell top bar with TopIcons extension, fine e.g. with default Adwaita, Numix-Frost, Arc and Paper GNOME Shell themes. Then again, this is with latest GNOME Shell 3.18.3.

In conclusion, it looks like I cannot reproduce this issue with Arch/GNOME, at least with `state-offline`.

**Comment by alefa** [comment]
Well, it's not a big deal, I can live with it. Since none of you can reproduce the issue, I guess the blurry look only occurs on high-resolution screens.

**Comment by bil-elmoussaoui** [comment]
I don't know if non monochrome icons are hardcoded. If they're you can try this as i mentionned before https://github.com/bil-elmoussaoui/Hardcode-Tray

**Comment by palob** [comment]
@alefa If your window scaling is set to 1 you're actually using your screen like a LoDPI one.
Aren't icons tiny this way?

**Comment by alefa** [comment]
@bil-elmoussaoui Thanks, I'll check it out!
@palob No, they're not tiny because the font scaling setting in Gnome Tweak Tool also affects icon size. I don't use window scaling because it only allows integer values. If I set window scaling to 2, everything becomes too big. With font scaling set to 1.6, icons and fonts have the right size for my liking. Here's a [screenshot](https://imgur.com/iCy1RKM).

**Comment by palob** [comment]
With a non-integer scaling factor you'll always get blurry icons in most cases.
It's still strange it looks like it does.
How does it look with font scaling set to 1?

I've got a similar issue with the VLC systray icon. With the TopIcons extension it looks similarly squashed (I didn't bother to make it use a proper panel icon in lieu of the Circle icon so far, just turned it off altogether):
![vlc](https://cloud.githubusercontent.com/assets/7050624/11875043/f6458890-a4e2-11e5-9854-1db5e9c603b8.png)

**Comment by alefa** [comment]
@palob Here's a [screenshot](https://imgur.com/wrgkjF3) with font scaling set to 1. So the squashing indeed seems to be caused by font scaling. The thing is, at least on my system almost all icons and other interface elements look fine with a non-integer scaling value, whereas they are either tiny or huge with integer scaling values. So I think I will keep using these settings until Linux gets proper support for non-integer scaling with Wayland.
By the way, the VLC systray icon on my system looks as in your screenshot.

**Comment by wa4557** [comment]
@palob seems to be the scaling parameter again... Are you opening a PR for this?

**Comment by palob** [comment]
Yeah, soon but I'll fiddle about some more :)

**Comment by dirtydancing** [comment]
I can confirm the squasing of the VLC icon (circle app icon) with TopIcons extension as well. In contrast, the VLC icon looks fine in the default hidden "legacy" tray in the lower left.

Font scaling is causing trouble with several "legacy" icons in the top bar with TopIcons at my end as well.

**Comment by palob** [comment]
@alefa Is the default non-Numix owncload icon always fine?

**Comment by dirtydancing** [comment]
At my end, the non-Numix owncloud icon also changes somewhat in the top bar with TopIcons when font scaling, but not as much as the Numix owncloud icon.

**Comment by palob** [comment]
But does it look as frazzled?

This very problem aside, the Numix icons for ``state-*`` suffer from misalignments.

**Comment by wa4557** [comment]
They are? (Ok the 16px sizes are misaligned but they are more like a gimmick and not really necessary)

I think it is most certainly the `scalable` part in the index.theme. Apart from pure GTK programs this seems to cause problems

**Comment by palob** [comment]
Ok, so I'll compile a PR changing all sub-48px icons to ``Type=Fixed``.
If we encounter any regressions, we can revert the respective sections.

**Comment by alefa** [comment]
@palob As far as I can see, the default owncloud status icon looks fine regardless of font scaling.

**Comment by palob** [comment]
Can you look up in which sizes it is present? PNG or SVG?

**Comment by alefa** [comment]
I can't seem to find the icons on my system, but in the source of the Debian package and in the owncloud client's github repository they're available as PNG and SVG ([github link](https://github.com/owncloud/client/tree/2.0.2/theme/white)).The PNG's have a maximum size of 256 pixel.

**Comment by palob** [comment]
OK, so maybe it's another case of an out-of-place 22/24px icon?

**Comment by alefa** [comment]
I'm using the git version of numix, but I haven't updated since I cloned the repository about a month ago. I'll do that tomorrow when I get back to my work computer.

**Comment by alefa** [comment]
So I'm now using the latest git version of Numix. Owncloud status icon still looks squashed and blurry.

**Comment by wa4557** [comment]
Has anyone tried out if the fix by @palob #951 fixes this issue as well?

**Comment by alefa** [comment]
Just pulled the newest version from GitHub. The owncloud status icon hasn't changed, so the issue is not fixed for me.

**Comment by wa4557** [comment]
You have to try this branch: https://github.com/palob/numix-icon-theme/tree/blurry-actions-fix

**Comment by alefa** [comment]
Ah, I see. I thought it had already been merged. Anyway, with @palob's branch the owncloud icon still looks the same as before.

**Comment by palob** [comment]
Maybe the issue is due to the missing owncloud icon sizes in Numix?

**Comment by wa4557** [comment]
There are no sizes missing... the icons are present in `16` `22` and `24` plus their respective `@2x` folders

**Comment by palob** [comment]
But owncloud have [those size suffix icons in their repo](https://github.com/owncloud/client/tree/2.0.2/theme/white).

**Comment by wa4557** [comment]
@alefa Is this still an issue? We changed the owncloud-icons a while back...

**Comment by alefa** [comment]
@wa4557 Sorry for the late reply, I've been away from my computer for a couple of weeks. Well, the icon is no longer blurry, but it is now so tiny that it looks quite [strange](https://imgur.com/a/TGNAH).

**Comment by wa4557** [comment]
Looks good to me. For XFCE tray icons seem to be smaller after all

**Comment by alefa** [comment]
I use XFCE on my other laptop (no HiDPI display). There the ownCloud client icon [looks much bigger](https://imgur.com/a/WcPqJ) in relation to other tray icons than on Gnome. Anyway, it's not a big deal, just my aesthetic impression. It doesn't affect the functionality of the tray icon.

**Comment by wa4557** [comment]
@Foggalong I guess this one can be closed

**Comment by Foggalong** [comment]
Not for good reasons though :cry:
